js/page.js

```diff
@@ -36,14 +36,18 @@
           window.map.mapPinsNode.appendChild(fragment);
         };
 
-        window.onChangeTypeFilterNode = function () { // очищаем карту и вызываем отрисовку отфильтрованных по типу пинов
-          window.map.clearMap();
-          renderMapPins(window.filter.filterMapPins(adObjectsArr));
+        window.onChangeFilterNode = function () { // очищаем карту и вызываем отрисовку отфильтрованных по типу пинов через дебаунс
+          var updateMapPins = window.debounce(function () {
+            window.map.clearMap();
+            renderMapPins(window.filter.filterMapPins(adObjectsArr));
+          });
+
+          updateMapPins();
         };
 
         renderMapPins(window.filter.filterMapPins(adObjectsArr)); // отрисовываем пины при фильтрах по умолчанию
 
-        window.filter.typeFilterNode.addEventListener('change', window.onChangeTypeFilterNode); // добавляем обработчик для фильтра по типу
+        window.filter.filterNode.addEventListener('change', window.onChangeFilterNode); // обработчик изменения формы
```

Comment:
Именование обработчиков должно быть `on + место + событие`. Проверь везде пож
```suggestion
        window.filter.filterNode.addEventListener('change', window.onFilterNodeChange); // обработчик изменения формы
```